GX-TimeFrequency/src/minimax_grids.F90

```diff
@@ -127,27 +166,41 @@ subroutine gx_minimax_grid(num_points, e_min, e_max, &
 
     ! Compute the actual weights used for the inhomogeneous cosine/ FT and check whether
     ! the two matrices for the forward/backward transform are the inverse of each other.
-    do j_point = 1, num_points
-       do i_point = 1, num_points
-          cosft_wt(j_point, i_point) = cosft_wt(j_point, i_point)*cos(tau_points(i_point)*omega_points(j_point))
-          cosft_tw(i_point, j_point) = cosft_tw(i_point, j_point)*cos(tau_points(i_point)*omega_points(j_point))
-          sinft_wt(j_point, i_point) = sinft_wt(j_point, i_point)*sin(tau_points(i_point)*omega_points(j_point))
-       end do
-    end do
+    if(.not.my_bare_cossine_weights) then
```

Comment:
A matter of taste, but why not switch this and have `if(my_bare_cossine_weights)`?